Orchestrator.py

```diff
@@ -128,29 +126,178 @@ def receive(self,frame):
         assert frame['frameType'] == self.FRAMETYPE_NOTIFICATION
 
         # shorthand
-        dotbot       = self.dotBotsView[frame['source']]
-        log.debug('dotbot {} was at ( {},{} ) '.format(dotbot, dotbot['x'], dotbot['y']))
+        dotBot       = self.dotBotsView[frame['source']]
 
         # update DotBot's position
         (newX, newY) = u.computeCurrentPosition(
-            currentX = dotbot['x'],
-            currentY = dotbot['y'],
-            heading  = dotbot['heading'],
-            speed    = dotbot['speed'],
+            currentX = dotBot['x'],
+            currentY = dotBot['y'],
+            heading  = dotBot['heading'],
+            speed    = dotBot['speed'],
             duration = frame['movementDuration'],
         )
 
+        # update explored cells
+        cellsExplored          = self._computeCellsExplored(dotBot['x'], dotBot['y'], newX, newY)
+        self.cellsExplored    += cellsExplored['cellsExplored']
+
+        if cellsExplored['nextCell'] and self._xy2cell(newX, newY) != (newX, newY):
+            self.cellsObstacle  += [cellsExplored['nextCell']]
+
+        # remove duplicate cells
+        self.cellsObstacle  = list(set(self.cellsObstacle))
+        self.cellsExplored = list(set(self.cellsExplored))
+
         # update dotBotsView
-        dotbot['x']       = newX
-        dotbot['y']       = newY
-        log.debug(f'dotbot {dotbot} is at ( {newX},{newY} ) ')
+        dotBot['x']       = newX
+        dotBot['y']       = newY
+        log.debug(f'dotbot {dotBot} is at ( {newX},{newY} ) ')
 
-        # pick new speed and heading for dotbot
+        # pick new speed and heading for dotBot
         (heading, speed)  = self._pickNewMovement(frame['source'])
-        dotbot['heading'] = heading
-        dotbot['speed']   = speed
+        dotBot['heading'] = heading
+        dotBot['speed']   = speed
+
+    #=== Map
+
+    def _computeCellsExplored(self, ax, ay, bx, by):
+        '''
+        find cells passed through on trajectory between points a and b
+        example input - output :
+            {
+            'in': {
+                'ax': 1.10,
+                'ay': 1.20,
+                'bx': 2.20,
+                'by': 1.01,
+            },
+            'out': {
+                'cellsExplored': [(1.00,1.00), (1.50, 1.00), (2.00, 1.00)],
+                nextCell: (2.50, 1.00)
+                }
+        }
+        '''
+        returnVal = []
+
+        # set start and stop cells
+        (startX, startY) = (ax, ay)
```

Comment:
since now the order of the parameters matters (it's not just a point A and B, it's a start a stop) just change the names of the params to startX, starty, stopX, stopY